packages/apps/sale/src/models/objectFields.ts

```diff
@@ -46,8 +46,39 @@ export const sale_modelAPI: ObjectFields = {
     saleProductMultipleQtyList: schemaContructor
       .array()
       .of(schemaContructor.subObject()),
-    productVariant: schemaContructor.subObject('name'),
-    parentProduct: schemaContructor.subObject('name'),
+    productVariantConfig: schemaContructor.subObject(),
+    productVariant: schemaContructor.subObject().concat(
+      schemaContructor.object({
+        productVariantAttr1: schemaContructor.subObject(),
+        productVariantAttr2: schemaContructor.subObject(),
+        productVariantAttr3: schemaContructor.subObject(),
+        productVariantAttr4: schemaContructor.subObject(),
+        productVariantAttr5: schemaContructor.subObject(),
+        productVariantValue1: schemaContructor.subObject(),
+        productVariantValue2: schemaContructor.subObject(),
+        productVariantValue3: schemaContructor.subObject(),
+        productVariantValue4: schemaContructor.subObject(),
+        productVariantValue5: schemaContructor.subObject(),
+      }),
+    ),
+    parentProduct: schemaContructor.subObject().concat(
+      schemaContructor.object({
+        name: schemaContructor.string(),
+      }),
+    ),
```

Comment:
I think you can leave it as it was before : `parentProduct: schemaContructor.subObject('name')`